src/Nncase.Importer/Onnx/Reshape.cs

```diff
@@ -13,6 +14,45 @@ public partial class OnnxImporter
         private Expr VisitReshape(in NodeProto op)
         {
             var (input, shape) = GetInputExprs(op, 0, 1);
+            var inputShape = F.Tensors.ShapeOf(input);
+            var shapeValue = ((TensorConst)shape).Value.ToArray<long>();
+            var actualShape = new Expr[shapeValue.Length];
+            var negAxis = shapeValue.Length;
+            for (int i = 0; i < actualShape.Length; i++)
+            {
+                if (shapeValue[i] == 0L)
+                {
+                    actualShape[i] = inputShape[i];
+                }
+                else if (shapeValue[i] == -1L)
+                {
+                    negAxis = i;
+                }
+                else
+                {
+                    actualShape[i] = shapeValue[i];
+                }
+            }
+
+            if (negAxis < shapeValue.Length)
+            {
+                Expr productOut = 1L;
+                for (int i = 0; i < shapeValue.Length; i++)
+                {
+                    if (i != negAxis)
+                    {
+                        productOut *= actualShape[i];
+                    }
+                }
+
+                Expr productIn = 1L;
+                for (int i = 0; i < input.CheckedShape.Rank; i++)
+                {
+                    productIn *= inputShape[i];
+                }
+
+                actualShape[negAxis] = productIn / productOut;
+            }
 
             // allowzero has been avaliable since opset 14
             var allowZero = GetBoolAttribute(op, "allowzero", false);
```

Comment:
you should remove this check when you support zero shape

Reply:
allowZero is not zero shape, they are different things